DOC Adds a cibuildwheel github action example to out of tree build docs (#5062)
pyodide/pyodide: docs/development/building-and-testing-packages.md

If your package is a pure Python package (i.e., if the wheel ends in
`py3-none-any.whl`) then follow the official PyPA documentation on building
[wheels](https://packaging.python.org/en/latest/tutorials/packaging-projects/#generating-distribution-archives)

> For binary packages, the manual steps are detailed below. In addition,
> [cibuildwheels](https://cibuildwheel.pypa.io/en/stable/) 2.19 or later provide
> support for building binary packages.

Reviewer: ```suggestion
For binary packages, the manual steps are detailed below. In addition,
[cibuildwheel](https://cibuildwheel.pypa.io/en/stable/) 2.19 or later provides
support for building binary wheels with Pyodide as a target.
```

work, so if you have a test runner that uses `subprocess` then it cannot be
used.

> ## Build Github actions examples

Reviewer: ```suggestion
## Github Actions build examples
```

> This is a complete example for building a Python wheel out of tree using

Reviewer: ```suggestion
This is a complete example for building a Pyodide wheel out of tree using
```

`pyodide build` directly:

```yaml

> runs-on: ubuntu-22.04

Reviewer: ```suggestion
runs-on: ubuntu-22.04 # or ubuntu-latest
```

I think we should specify that the `ubuntu-22.04` image is not a hard requirement

  steps:
  - uses: actions/checkout@v4
  - uses: actions/setup-python@v5
    with:

>        python-version: 3.12

Reviewer: ```suggestion
       python-version: 3.12.1
```
